### DISCUSSION TITLE
GH-Actions step hangs for ever on ubuntu-latest runner without step output

### DISCUSSION THREAD

**Discussion by g-bougard** [community/community#33845] [Actions] [answered]
Hi there,

since yesterday, I was trying to optimize a little my test jobs from this [glpi-agent workflow](https://github.com/glpi-project/glpi-agent/blob/develop/.github/workflows/glpi-agent-ci.yml).

Since last 2 commits, I have the linux-ubuntu-test which is stucked on `Run tests` step and the raw log just shows few lines:
```
2022-09-22T13:02:27.8468913Z Requested labels: ubuntu-latest
2022-09-22T13:02:27.8468973Z Job defined at: glpi-project/glpi-agent/.github/workflows/glpi-agent-ci.yml@refs/heads/develop
2022-09-22T13:02:27.8468996Z Waiting for a runner to pick up this job...
2022-09-22T13:02:29.2626815Z Job is waiting for a hosted runner to come online.
2022-09-22T13:02:33.0061496Z Job is about to start running on the hosted runner: Hosted Agent (hosted)
```
And passed previous steps are not included in raw log.
I think the raw log is really updated when the job is finished.
Here is the link to a job still in running state at the time of this writing for more than 2 hours:
https://github.com/glpi-project/glpi-agent/actions/runs/3105081687

The `Run tests` step is only starting `make test`...

I tried to cancel the job and re-run the failed job after a first stuck for more than an hour. And the new run passed the first step in the same way and stay stuck at the same step still for more than an hour.

The only differences between the last commit where the job didn't failed and the one it is stuck are:
1. I removed my try to cache built perl libraries from CPAN as it seems this is not permitted to cache system folders under /usr/local
2. I changed my step where I was only installing via apt `cpanminus` and `libmodule-install-perl` packages to install as much as possible perl modules packages available in distro to not have to build then from cpan.

The concerned steps are running before the stuck step and everything seems fine.

Is there any temporary issues with GH Actions which prevent my step to start ?

I'll try to revert my changes to what was working yesterday, but I really don't see anything wrong in my changes.

**Comment by g-bougard** [comment]
Well, aborting one of the stuck job I finally see the log and understood one of our tests is failing, probably due to a perl package issue which wasn't a problem while the module was installed from perl CPAN.

The bug involves one test which is trying to run a http daemon to test our http API. This fork a process which is unfortunately not stopped and the perl test waits forever for this process to stop. I can figure out this issue.

So my last concern here is why the step output is not shown in that case.

**Comment by g-bougard** [comment, marked as answer]
I think I figured out why output was not showing:
1. if you follow the workflow output since it starts, you'll see the last error,
2. if you connect to the workflow after the last line was sent to the output, you won't see the lines from the output as only new lines are shown.

In the second case, maybe last 10 lines could be shown before new lines to be able to identify a stuck case like mine.

**Reply by farhaven** [reply]
Thanks for confirming it's not just me going crazy. I have the same problem here. @github-staff is there a way to keep the logs from failed tasks, even if the runner they belong to has crashed/disappeared/whatever?

The way it's done right now isn't that useful, because when everything is OK and working fine, I'm not really interested in the output. I'm interested if it fails, for whatever reason (not just "command exited with status 1", but also "runner went away and never came back"), and "just keep the tab open" isn't really an option (some builds run for a long time, or overnight, or I just don't get around to opening the tab after a push, ...)
